### ISSUE TITLE
Cross entropy with sequence log_softmax not (almost) equal to 1

### ISSUE THREAD

**Issue by davebulaval** [pytorch/pytorch#27626]
## ❓ Is this a feature or a bug

When using the Cross entropy with K-target class in a sequence application, is it expected to squeeze before calling the loss or we are suppose to just past a vector of minibatch_size, sequence_length, K-tag ?

Because, in this line of code `nll_loss(log_softmax(input, 1), target, weight, None, ignore_index, None, reduction)`  the `log_softmax` step is giving me prob vector greater than 1 or not (almost) equal to 1. 
> After the exponantial application for sure.

Here an example of my situation, when 
```
y_target = [0, 1, 1, 1, 1, 1, 1, 7, 3, 3, 4, 5]

y_predict = tensor([[[-3.2910, -3.7040, -3.9663, -4.1059, -4.1012, -4.1368, -4.1158,
              -4.2506, -4.3760, -4.1655, -3.7205, -2.8966],
             [8.8342, 9.8005, 9.9371, 9.5386, 9.1664, 8.5789, 7.4153,
              3.6922, 1.4415, -4.2969, -9.6262, -9.6677],
             [4.8800, 3.9539, 3.4548, 3.2552, 3.2441, 3.1305, 3.0244,
              3.4114, 2.4158, 0.8764, -1.0089, -1.9860],
             [0.9770, 0.4299, 0.5240, 0.6457, 0.3984, 0.4953, 0.4214,
              0.4627, 3.8482, 5.2386, 3.0858, 1.8424],
             [-7.9146, -7.7125, -7.8830, -7.8319, -7.5219, -7.4129, -6.9861,
              -5.2630, -5.4942, 1.5527, 8.9790, 4.6239],
             [-11.0455, -10.9685, -10.4619, -9.8926, -9.7860, -9.2574, -8.6142,
              -6.8966, -3.1693, -1.9010, 2.4475, 10.5265],
             [3.2678, 3.2880, 3.1630, 3.0535, 3.2340, 3.1786, 3.6118,
              3.5331, 1.1905, 0.1634, -0.8760, -2.4547],
             [1.0813, 1.1683, 1.2462, 1.3132, 1.4225, 1.4196, 1.5218,
              1.9667, 1.2652, 0.6920, -0.1019, -0.8134],
             [0.9259, 1.0986, 1.2023, 1.2958, 1.3188, 1.3190, 1.3048,
              1.1060, 1.0091, 0.1047, -0.7875, -0.9094]]])

log_softmax(y_predict, 1) # vector prob not (almost) equal to 1
log_softmax(y_predict, 2) # vector (almost) equal to 1
```

**Comment by zou3519** [comment]
I don't know the expected behavior of cross entropy. It would be good to get someone who knows the behavior to chime in.

**Comment by davebulaval** [comment]
It's seem odd to me the output of the `log_softmax`. But as I was trying to work my head around it i'm not sure of the expected behavior when using sequence.

**Comment by gchanan** [comment]
I'm going to close this, because it's really a question for the forums (Please use https://discuss.pytorch.org (https://discuss.pytorch.org/) for questions).  If you find it's a bug, please reopen this.
